src/main/scala/org/constellation/checkpoint/CheckpointAcceptanceService.scala

```diff
@@ -522,10 +510,19 @@ object CheckpointAcceptanceService {
           case (hash, txs) =>
             txChainService
               .getLastAcceptedTransactionRef(hash)
-              .map(txs.headOption.map(_.lastTxRef).contains)
+              .map {
+                latr =>
+                  txs.foldLeft(LastTransactionRef.unapply(latr)) {
+                    case (maybePrev, tx) =>
+                      for {
+                        prev <- maybePrev
+                        txPrev <- LastTransactionRef.unapply(tx.lastTxRef) if prev == txPrev
+                      } yield (tx.hash, tx.ordinal)
```

Comment:
Do we need to use the unapply instead of just LastTransactionRef directly? I guess you use it to get an Option but maybe you could just explicitly wrap LastTransactionRef in an Option with .some and then operate on LastTransactionRef?

Reply:
so:
`tx.foldLeft(latr.some)`
and then
`lastTxRef = if (tx.lastTxRef == prev) LastTransactionRef(tx.hash, tx.ordinal).some else None
} yield lastTxRef`

Reply:
I use unapply to get a tuple `(String, Long)` from `tx.lastTxRef` which I can compare in next iteration with a tuple `(String, Long)` coming from an actual previous transaction `(tx.hash, tx.ordinal)`. I can use `LastTransactionRef` case class directly instead of a tuple `(String, Long)` if you think it will make it more readable.